nhrpd/vici.c

```diff
@@ -469,16 +469,49 @@ static void vici_register_event(struct vici_conn *vici, const char *name)
 	vici_submit(vici, obuf);
 }
 
+static bool vici_charon_filepath_done;
+
+static char *vici_get_charon_filepath(void)
+{
+	static char buff[1200];
+	FILE *fp;
+	char *ptr = buff, *ptr2;
+	char line[1024];
+
+	if (vici_charon_filepath_done)
+		return ptr;
+	fp = popen("ipsec --piddir", "r");
+	if (!fp) {
+		zlog_err("%% Failed to retrieve charon file path.");
+		return NULL;
+	}
+	while (fgets(line, sizeof(line), fp) != NULL) {
```

Comment:
what's happening here? it looks as if we'll read lines, then each line will overwrite at the beginning of 'buff'. but 'ptr' is being advanced down buff each iteration into ... unknown territory? I sort of get returning 'buff', where the last thing you read will have been formatted - but why mess with 'ptr' at all? you either have something in 'buff', or you need to put something into 'buff' ?

Reply:
simplified algorithm and added an explanation